java/ql/src/utils/model-generator/CaptureSummaryModels.ql

```diff
@@ -0,0 +1,280 @@
+/**
+ * @name Capture summary models.
+ * @description Finds applicable summary models to be used by other queries.
+ * @id java/utils/model-generator/summary-models
+ */
+
+import java
+import semmle.code.java.dataflow.TaintTracking
+import semmle.code.java.dataflow.internal.DataFlowImplCommon
+import semmle.code.java.dataflow.internal.DataFlowNodes
+import semmle.code.java.dataflow.internal.DataFlowPrivate
+import semmle.code.java.dataflow.InstanceAccess
+import ModelGeneratorUtils
+
+string captureFlow(Callable api) {
+  result = captureQualifierFlow(api) or
+  result = captureParameterFlowToReturnValue(api) or
+  result = captureFieldFlowIn(api) or
+  result = captureParameterToParameterFlow(api) or
+  result = captureFieldFlow(api)
```

Comment:
Taking a step back, I think all four of these could be captured by a single configuration from `ParameterNode` (i.e. incl. `this`) to `ReturnNodeExt`. Additional steps should likely then allow both `read` and `store` steps - ideally in the order read-read-read-store-store-store, i.e. all read-as-taint occurring before the store-as-taint steps. The relevantType constraint could be added as a barrier/sanitizer.

Reply:
But we can postpone such a simplification to a follow-up PR.